test/sframe/sframe_bench_aggregate.cpp

```diff
@@ -0,0 +1,159 @@
+#include <core/storage/sframe_data/groupby_aggregate_operators.hpp>
+#include <core/storage/sframe_data/sframe.hpp>
+#include <core/storage/sframe_data/sframe_constants.hpp>
+#include <iostream>
+
+namespace turi {
+
+using row_gen_func_t = std::function<std::vector<flexible_type>(size_t)>;
+
+/* generate sframe */
+/* I need to make this a fixture */
+static sframe bench_test_sframe_generator(
+    const std::vector<std::string>& column_names,
+    const std::vector<flex_type_enum>& column_types, size_t nrows,
+    row_gen_func_t next_row) {
+  if (column_types.size() != column_names.size())
+    log_and_throw("column_types size mismatches with column_names size");
+
+  if (next_row(0).size() != column_types.size())
+    log_and_throw(
+        "row size mismatches with the output of row generate function");
+
+  // calculate intervals
+  size_t nthreads = thread::cpu_count();
+  size_t interval = nrows / nthreads;
+  std::vector<std::pair<size_t, size_t>> write_intervals;
+  for (size_t ii = 0; ii < nthreads; ii++) {
+    if (ii) {
+      auto start = write_intervals.back().second;
+      write_intervals.emplace_back(start, start + interval);
+    } else {
+      write_intervals.emplace_back(0, interval);
+    }
+  }
+  write_intervals.back().second = nrows;
+
+  // consturct sframe
+  sframe out;
+  out.open_for_write(column_names, column_types, "", nthreads);
+  std::vector<turi::sframe_output_iterator> write_iters;
+  write_iters.resize(nthreads);
+  for (size_t ii = 0; ii < nthreads; ii++)
+    write_iters[ii] = out.get_output_iterator(ii);
+
+  parallel_for(0, write_iters.size(), [&](size_t ii) {
+    size_t start = write_intervals[ii].first;
+    size_t end = write_intervals[ii].second;
+    auto out_iter = write_iters[ii];
+    while (start < end) {
+      *out_iter = next_row(start);
+      start++;
+    }
+  });
+
+  // finish writing
+  out.close();
+
+  return out;
+}
+
+static void bench_test_aggreate(const sframe& in_sf,
+                                std::shared_ptr<group_aggregate_value> op,
+                                size_t nthreads, bool debug_print = false) {
+  if (nthreads == 0 || nthreads > thread::cpu_count())
+    log_and_throw("invalid thread count");
+
+  // set up groupby_aggregate_impl
+  auto& pool = thread_pool::get_instance();
+  size_t old_pl_sz = pool.size();
+
+  bool need_to_set_thread_size = nthreads != pool.size();
+  if (need_to_set_thread_size) {
+    pool.resize(nthreads);
+  }
+
+  logstream(LOG_INFO) << "Bench test groupby with count with " << nthreads
+                      << "threads." << std::endl;
+
+  turi::timer ti;
+  const std::string& key = in_sf.column_names().front();
+  sframe out_sf =
+      groupby_aggregate(in_sf, {key}, {"my_cnt"}, {{{}, op}});
+
+  logstream(LOG_INFO) << "Bench test groupby with count finished in " << ti.current_time()
+            << std::endl;
+
+  // restore the default thread pool size
+  if (need_to_set_thread_size) pool.resize(old_pl_sz);
```

Comment:
This can be done externally using the OMP_NUM_THREADS environment variable, so no need to programmatically do that here.  Just use the default thread pool size.